app/Http/Controllers/PartyController.php

```diff
@@ -87,10 +87,9 @@ public function index($group_id = null)
             $upcoming_events_in_area = null;
         } else {
             $upcoming_events = Party::upcomingEvents()->where('users_groups.user', Auth::user()->id)
-                ->take(3)
                 ->get();
 
-            $past_events = Party::UsersPastEvents([auth()->id()])->paginate(10);
```

Comment:
Pulling back all results has a bit of a performance hit on the /party page - logged in as a user with 263 past events, it takes 7 seconds to render on my laptop, so will presumably also have the performance issue James logged for his mobile.  Not sure if there's anything we can do here right now?  But we'll need to address this soon I think.